### ISSUE TITLE
Typesafe selectors

### ISSUE THREAD

**Issue by alex-deas** [erikras/redux-form#3099]
### Are you submitting a **bug report** or a **feature request**?
Bug

### What is the current behavior?
```javascript
const Form = connect(state => ({
  submitErrors: getFormSubmitErrors('foo')(state), // submitErrors will be an Object or undefined
})(FormComponent);
```

### What is the expected behavior?
This is expected to be an empty object rather than undefined

[Docs](http://redux-form.com/6.8.0/docs/api/Selectors.md/#-getformsyncwarnings-formname-string-returns-state-formsyncwarnings-object-): `getFormSubmitErrors(formName:String) returns (state) => formSubmitErrors:Object`

### What's your environment?
v6.8

**Comment by erikras** [comment]
Fix published in [`v7.2.0`](https://github.com/erikras/redux-form/releases/tag/v7.2.0).

**Comment by oliverlaz** [comment]
Hi @erikras, hi @alex-shatalov,

Thanks for aligning the implementation with the documentation :)

However, this change eventually breaks userland code where weak truthiness checks are used.
For example, the following code is broken with v7.2.0:
```javascript
const errors = getFormSyncErrors(form)(state);
if (errors) {
  // always true with v7.2.0
} else {
  // do something
}
```

I think it would be good if this change is published somewhere, maybe in the Release Notes.
